x-pack/plugins/enterprise_search/server/lib/engines/available_indices.ts

```diff
@@ -11,10 +11,24 @@ export const availableIndices = async (
   client: IScopedClusterClient,
   indices: string[]
 ): Promise<string[]> => {
-  if (await client.asCurrentUser.indices.exists({ index: indices })) return indices;
+  const closedIndicesList = await closedIndices(client);
 
-  const indicesAndExists: Array<[string, boolean]> = await Promise.all(
-    indices.map(async (index) => [index, await client.asCurrentUser.indices.exists({ index })])
+  const indicesExistsAndOpen: Array<[string, boolean]> = await Promise.all(
```

Comment:
question: do we actually need to check whether these indices actually exist? We should basically never have a non-existent index at this point.

Second point: if we do need to do this, we should not do it this way. What you're doing now is calling `client.asCurrentUser.indices.exists({ index })` for every index in this list (which is potentially very large). That's a *lot* of concurrent requests that can stall or crash (due to OOM) either Elasticsearch or Kibana. You should be making a single call for all these indices, and filter out the closed indices from the list before doing that.

Reply:
> question: do we actually need to check whether these indices actually exist? We should basically never have a non-existent index at this point.

The exist API call was added to make sure that index is not deleted. 

> You should be making a single call for all these indices, and filter out the closed indices from the list before doing that.

Yeah, based on the slack [discussion](https://elastic.slack.com/archives/C9BA8D0P4/p1685573272118129), the problem here is that current GET API does not filter out closed indices when there is an associated alias. We could certainly get all open indices with` GET *?expand_wildcards=open` but 
we would still need another API call to make sure that index is not closed may adding `*.settings.index.verified_before_close` to filter_path